src/hip_module.cpp

```diff
@@ -40,6 +40,8 @@ THE SOFTWARE.
 #include "hip_hcc_internal.h"
 #include "trace_helper.h"
 
+//TODO Add support for multiple modules
+static std::unordered_map<std::string, uintptr_t> coGlobals;
```

Comment:
I know this is not new here but we need to remove this global or protect access from multiple threads.